crates/noirc_evaluator/src/ssa/block.rs

```diff
@@ -0,0 +1,207 @@
+use super::{code_gen::IRGenerator, node};
+use arena::Index;
+use std::collections::{HashMap, VecDeque};
+
+#[derive(PartialEq, Debug)]
+pub enum BlockType {
+    Normal,
+    ForJoin,
+}
+#[derive(Debug)]
+pub struct BasicBlock {
+    pub idx: arena::Index,
```

Comment:
Because instruction and block are so different in nature I do not feel the need for the distinction on the index name and rather think it is coherent to have an index just being called like that.

Reply:
My thinking is that because they are so different in nature this difference should be reflected in their types as well. It also follows other parts of our codebase where we have different types for ExprID, StmtId, FuncId, etc., which are all arena indices under the hood.

Reply:
I discussed with Kev and he agrees with you, however I tried and finally I prefer to delay the change after the merge because it requires many modifications. If it is better to do it before merging, let me know and I will do it.